# **World Retail Data**

In [ ]:
import pandas as pd

In [ ]:
file = "/content/drive/MyDrive/Praktikum_AVD/business_understanding_AVD/online_retail_real_world.csv"
df = pd.read_csv(file)
pd.set_option('display.max_rows', None)
df

,OrderID,CustomerID,ProductName,Brand,Raw_Weight,Country,OrderDate,UnitPrice
0,REC-861088,6120,Perly,Perly,100 g,"Morocco,United States",2024-02-22,13.18
1,REC-900952,12550,Prince Goût Chocolat,LU,300 g,"Algeria,Belgium,France,French Polynesia,German...",2024-11-13,1.50
2,REC-595965,6705,Excellence Noir Prodigieux 90% Cacao,"Lindt EXCELLENCE,Lindt",100g,"Argelia,Austria,Bélgica,Bulgaria,Canadá,Repúbl...",2025-02-02,12.99
3,REC-942653,8645,Tonik,عربي,22 g,Maroc,2024-12-11,14.82
4,REC-885998,14430,Sésame,Gerblé,230g,"Belgium, Bulgaria, France, en:switzerland",2025-08-21,18.40
5,REC-674324,6760,Шоколад 85% какаова маса,J. D. Gross,125g,"Austria,Belgium,Bulgaria,Estonia,Finland,Franc...",2024-09-08,19.16
6,REC-867660,8420,CRISTALINE Eau De Source 0.5L,Cristaline,500 ml,"Belgium,Côte d'Ivoire,France,Germany,Luxembour...",2024-10-10,3.64
7,REC-448132,9144,Henry’s,Henry's,42g,Maroc,2024-12-30,17.59
8,REC-202400,10397,Milk chocolate bar,NaN,150 g,"Algeria,Cameroon,France,Morocco,Spain",2024-08-04,11.08
9,REC-386572,13888,Organic 70% Dark Chocolate Bar,Green & Black's,90g,"France,Irlande,Suède,Royaume-Uni",2025-10-15,12.14


## **Business Understanding**


### **business objective**

**Isi Dataset:**
Dataset ini merupakan catatan transaksi retail online yang mencakup berbagai produk (seperti cokelat, biskuit, dan makanan ringan) dari berbagai merek global. Data ini mencatat detail pesanan, ID pelanggan, berat produk, asal negara, hingga harga per unit.

**Masalah yang ada:**
Meskipun terdapat data transaksi dalam dataset ini, perusahaan mungkin akan mengalami kesulitan menentukan strategi harga dan manajemen stok yang efisien. Tanpa analisis yang mendalam, sulit untuk mengetahui produk mana yang paling laris atau jarang dibeli, bagaimana distribusi berat produk memengaruhi harga, atau di negara mana produk tersebut paling banyak tersedia.

**Tujuan analisis:**
adapun tujuan analisis ini dilakukan adalah untuk membantu perusahaan mengetahui produk mana yang laris di pasaran apa saja yang sering dibeli atau produk mana yang jarang dipesan kemudian membantu perusahaan mengetahui target pasar mereka nantinya

# **assess situation**

**Kondisi Dataset:**

-Format Berat yang Tidak Konsisten: Kolom Raw_Weight berisi teks seperti "150 g", "75 g", "170g", hingga "3".


-Data Geografis yang Kompleks: Kolom Country berisi daftar banyak negara dalam satu baris (misal: "France, Italy, Spain"). Ini menyulitkan jika kita ingin melakukan filter per negara secara spesifik.

-Skala harga yang tidak Teratur:Urutannya acak jadi harga tidak ter-urut berdasarkan yang paling mahal atau paling murah

-data tidak terdefinisi: dalam beberapa kolom atau baris terdapat teks NaN

## **analytic goals**

**Untuk mengatasi beberapa masalah di atas, berikut ini adalah target analisis yang akan dilakukan:**

-konversi ke jenis satuan yang sama: Mengonversi Raw_Weight menjadi angka numerik murni contohnya "gram" agar bisa dianalisis hubungannya dengan harga.

-Market Reach Mapping: Memecah data negara untuk melihat sebaran jangkauan produk secara global.

## **Project Plan**

**Berikut adalah langkah-langkah kerja yang akan diambil:**

Data Discovery (Memahami Data):


-Memeriksa tipe data setiap kolom dan menghitung data yang hilang atau NaN.


**Data Cleaning (Pembersihan):**

-Mengubah jenis satuan "g", "kg", atau "100g" dari kolom berat dan mengubahnya ke tipe yang sama.


-Mengisi atau mengategorikan merek yang hilang sebagai "Unknown".


-Memastikan tanggal order terbaca sebagai format tanggal yang benar.


-Exploratory Data Analysis (EDA): Membuat visualisasi distribusi harga, berat produk, dan frekuensi pesanan per bulan, agar rapi dan urut


-Reporting & Insights: Menyusun kesimpulan mengenai produk mana yang paling potensial untuk dipromosikan lebih lanjut.